processor/processor.go

```diff
@@ -34,6 +35,7 @@ type Processor interface {
 type TraceProcessor interface {
 	consumer.TraceConsumer
 	Processor
+	AddSupportExtensions(exts ...extension.SupportExtension) error
```

Comment:
This is a breaking change to an exported interface. We cannot do this without announcing first and making sure everyone is prepared.

I suggest you break this PR down into 2 parts:
1. Changes needed to core (such as this particular change). This needs to be discussed and agreed with all maintainers before moving forward. Preference will be given to enhancements that do not require breaking exporter interfaces. The barrier to any breaking changes is high, so please see if you can find a way to avoid it.

2. The rest of the implementation for the particular feature that you suggest. Note that this may not belong to this repository at all and may be better suited to either contrib if it is deemed useful for sufficiently large number of users or in your own repository.

Reply:
@annanay25 On start a component is passed a `Host` interface.  Perhaps we could add a way to request "support extensions" through this host?  

 A completely non breaking, but kind of opaque way would be through `Host.Context() context.Context`.